### ISSUE TITLE
git push fails with LFS

### ISSUE THREAD

**Issue by dg9ngf** [github/git-lfs#1089]
I have set up LFS with the commands from the LFS website in an existing repository with a GitLab remote. `git lfs ls-files` shows that one file is tracked by LFS. The push however fails completely now:

    D:\Projekte\Tests\GitLabPlayground>git push gitlab
    Git LFS: (0 of 1 files) 0 B / 672.83 KB
    exit status 1
    exit status 1
    error: failed to push some refs to 'git@db-gitlab:team/Playground.git'

What's wrong here?

```
D:\Projekte\Tests\GitLabPlayground>git lfs env
git-lfs/1.1.2 (GitHub; windows amd64; go 1.6; git bf6a6c9)
git version 2.7.2.windows.1

Endpoint (gitlab)=https://db-gitlab/team/Playground.git/info/lfs (auth=none)
  SSH=git@db-gitlab:team/Playground.git
LocalWorkingDir=D:\Projekte\Tests\GitLabPlayground
LocalGitDir=D:\Projekte\Tests\GitLabPlayground\.git
LocalGitStorageDir=D:\Projekte\Tests\GitLabPlayground\.git
LocalMediaDir=D:\Projekte\Tests\GitLabPlayground\.git\lfs\objects
TempDir=D:\Projekte\Tests\GitLabPlayground\.git\lfs\tmp
ConcurrentTransfers=3
BatchTransfer=true
GIT_SSH=C:\Program Files\TortoiseGit\bin\TortoiseGitPlink.exe
git config filter.lfs.smudge = "git-lfs smudge %f"
git config filter.lfs.clean = "git-lfs clean %f"
```

**Comment by anatolyborodin** [comment]
You need to use HTTPS repository. Git LFS works over HTTP AFAIR.

**Comment by dg9ngf** [comment]
Does it have to be https or will http do? This is an intranet server that can't have a publicly trusted certificate.

And can LFS support SSH, too? Pushing to SSH with a key is easy but entering username and password every single time is very annoying. SVN remembers that but Git doesn't. It seems Git only *wants* to be used over SSH not HTTP. So Git LFS wouldn't fit well to Git.

**Comment by anatolyborodin** [comment]
Hm, according to https://github.com/github/git-lfs/tree/master/docs/api Git LFS can work with SSH, but the server should support it (`git-lfs-authenticate` etc). GitLab doesn't yet: http://doc.gitlab.com/ce/workflow/lfs/manage_large_binaries_with_git_lfs.html . GitHub does.

> ... but entering username and password every single time is very annoying.

Have you seen https://help.github.com/articles/caching-your-github-password-in-git/ ?

**Comment by anatolyborodin** [comment]
@dg9ngf see #1047 btw

**Comment by anatolyborodin** [comment]
@dg9ngf there is a workaround for gitlab:

`git config remote.origin.lfsurl "https://gitlab.com/USER/REPO.git/info/lfs"`

^ note the **/info/lfs** part!

Worked for me.

See https://github.com/github/git-lfs/blob/master/docs/spec.md#the-server

**Comment by technoweenie** [comment]
Cool, looks like this is resolved.

**Comment by chipmunklove** [comment]
@anatolyborodin you're awesome! 💃

**Comment by kemin711** [comment]
I have a simple git and local working on the same machine.  No remote protocol.  Got two branches. Initially worked fine. The I got problems to push. For the first few times, I did delete then add back works. Now this hack stopped working.  git commit -a contain the following information:
`On branch master
Your branch is ahead of 'origin/master' by 5 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
=======
git lfs push --all origin
EOFoading LFS objects: 100% (33/33), 3.4 GB | 0 B/s    
================
git push --all origin
EOFoading LFS objects: 100% (1/1), 3.7 MB | 0 B/s                                                                               
error: failed to push some refs to '/home/git/dspkg'
`
Looks my trouble is not related to lfs.  Tried many methods, all failed.  Must be some of my config file is bad?  Appreciate help. No I could not push any more.
